lib/sidekiq/grouping/redis.rb

```diff
@@ -141,7 +141,10 @@ def delete(name)
       end
 
       def remove_from_pending(name, batch_name)
-        redis { |conn| conn.zrem(pending_jobs(name), batch_name) }
+        redis do |conn|
+          conn.zrem(pending_jobs(name), batch_name)
+          conn.del(batch_name)
```

Comment:
We could pipeline this to avoid the extra round-trip to redis.

Reply:
Good point. That should be fixed now.

Reply:
Technically, multi is for redis transactions. I'm not sure whether it has any overhead if you're not calling watch (necessary to make it transactional). It looks like there's precedent for (mis?)using this though, so it's fine with me.

Reply:
Actually, this makes me wonder whether a call to pipeline allows interleaving commands in the Redis server. If it does, then the semantics might be different since multi is guaranteed to be atomic. I think I tried to find that out before and didn't find the answer. Interleaving would be fine here though.